cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -949,10 +958,16 @@ impl ManagedEnvironment {
         Ok(env)
     }
 
-    pub fn push(&mut self, force: bool) -> Result<(), ManagedEnvironmentError> {
+    pub fn push(&mut self, flox: &Flox, force: bool) -> Result<(), ManagedEnvironmentError> {
         let project_branch = branch_name(&self.pointer, &self.path);
         let sync_branch = remote_branch_name(&self.pointer);
 
+        // ensure the environment builds before we push it
```

Comment:
```suggestion
        // Ensure the environment builds before we push it
        // If a user pulls an environment that is broken on their system, we may
        // create a "broken" generation.
        // That generation could have a divergent manifest and lock,
        // or it could fail to build.
```

Reply:
Why would you add _pull_ related comments here? Looks a bit out of context..

Reply:
More clear now? I was trying to think through why we'd ever need to have this check here

Reply:
I see, yeah i tumbled over the "pulling" bit just now, finding it rather weird to create a broken generation.. I guess that could be addressed by #681, i.e. by _not_ creating a generation in the first place...

Reply:
Yeah it's a bit weird, but I think it's what https://github.com/flox/flox/issues/823 is asking for? Unless do you read that differently?

Reply:
yeah, i think so